packages/core/admin/admin/src/pages/SettingsPage/pages/Roles/EditPage/hooks/usePermissionsDataManager.ts

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+
+// Note: I had to guess most of these types based on the name and usage, but I actually don't
+// know if they are correct, because the usage is very generic. Feel free to correct them if
+// they create problems.
+export interface PermissionsDataManagerContextValue {
+  availableConditions: unknown[];
+  modifiedData: object;
+  onChangeCollectionTypeLeftActionRowCheckbox: (
+    pathToData: string,
+    propertyName: string,
+    nam: string,
+    value: unknown
+  ) => void;
+  onChangeConditions: (conditions: object) => void;
+  onChangeSimpleCheckbox: (target: { name: string; value: unknown }) => void;
+  onChangeParentCheckbox: (target: { name: string; value: unknown }) => void;
+  onChangeCollectionTypeGlobalActionCheckbox: (
+    kind: string,
+    actionId: string,
+    value: unknown
+  ) => void;
+}
+
+export const PermissionsDataManagerContext = React.createContext({});
```

Comment:
```suggestion
export const PermissionsDataManagerContext = React.createContext<PermissionsDataManagerContextValue | null>(null);
```

suggestion only, but you've not added the type here so the hook will not have anything you've written above :)